specs/EnhancedSecurityMode.md

```diff
@@ -0,0 +1,468 @@
+        defaultInput: "");
+    if (dialog.ShowDialog() == true)
+    {
+        List<string> uris = WebViewProfile.EnhancedSecurityModeEnforceList;
+        List<string> inputList = dialog.Input.Text.Split(";").ToList();
+
+        foreach(string uri in inputList)
+        {
+            uris.Add(uri);
+        }
+                
+        MessageBox.Show("Enhanced Security Mode Enforce List is updated successfully",
+            "Enhanced Security Mode Enforce List", MessageBoxButtons.OK);
+    } 
+}
+```
+```cpp
+void SettingsComponent::GetEnhancedSecurityModeEnforceList() 
+{
+    wil::com_ptr<ICoreWebView2_13> webView2_13;
+    webView2_13 = m_webView.try_query<ICoreWebView2_13>();
+
+    if (webView2_13)
+    {
+        wil::com_ptr<ICoreWebView2Profile> profile;
+        CHECK_FAILURE(webView2_13->get_Profile(&profile));
+
+        auto profileStaging2 = profile.try_query<ICoreWebView2StagingProfile2>();
+        if (profileStaging2)
+        {
+            UINT32 counts = 0;
+            LPWSTR* uris;
+            profileStaging2->GetEnhancedSecurityModeEnforceList(&counts, &uris);
+            std::wstring s = L""; 
+            for (unsigned int i = 0; i < counts; ++i)
+            {
+               s += uris[i];
+               s += L"\n";
+            }
+
+            m_appWindow->AsyncMessageBox(s.c_str(), L"Enhanced Security Mode Enforce List");
+        }
+    }
+}
+
+void SettingsComponent::SetEnhancedSecurityModeEnforceList()
+{
+    wil::com_ptr<ICoreWebView2_13> webView2_13;
+    webView2_13 = m_webView.try_query<ICoreWebView2_13>();
+
+    if (webView2_13)
+    {
+        wil::com_ptr<ICoreWebView2Profile> profile;
+        CHECK_FAILURE(webView2_13->get_Profile(&profile));
+
+        auto profileStaging2 = profile.try_query<ICoreWebView2StagingProfile2>();
+        if (profileStaging2)
+        {
+            TextInputDialog dialog(
+                m_appWindow->GetMainWindow(), L"Add to Enhanced Security Mode Enforce List", L"List:",
+                L"Enter URI filters to enforce, separated by semicolons.", L"");
+            if (dialog.confirmed) {
+                // forming array
+                std::wstringstream ss(dialog.input);
+                std::wstring uri;
+                std::vector<std::wstring> uris;
+ 
+                while (std::getline(ss, uri, L';')) {
+                    uris.push_back(uri);
+                }
+ 
+                std::vector<LPCWSTR> uris_cstr;
+                for (const auto& uri : uris) {
+                    uris_cstr.push_back(uri.c_str());
+                }
+
+                CHECK_FAILURE(profileStaging2->SetEnhancedSecurityModeEnforceList(
+                    static_cast<uint32_t>(uris_cstr.size()), uris_cstr.data()));
+                MessageBox(nullptr, L"Enhanced Security Mode Enforce List is updated successfully",
+                    L"Enhanced Security Mode Enforce List", MB_OK);
+            }
+        }
+    }
+}
```

Comment:
The sample code should show the end dev how to perform a real scenario they may want to do in their app and in the way we imagine the feature should be used. In this sample code it looks like you let the end user edit everything which makes it hard to tell how to use the feature especially hard to tell what the form is of the strings added to the ESM lists.

I would guess that one scenario for the feature is the feature is turned on, its set to strict, and a URI filter for some things that are trusted and known to have perf issues are added. I would think that's the sort of common / recommended case for this feature? If so, then a sample should do that.